### ISSUE TITLE
Limit high-level, variable-length newtype's input to isize::MAX

### ISSUE THREAD

**Issue by brycx** [brycx/orion#130]
Zeroize seems to require that `Vec`s and slices have a length less than or equal to `isize::MAX`. Investigate where these limits are relevant in Orion, such as newtypes or allocations with zeroized data.

If there are places where this upper limit can be violated, we have to restrict it. This could possibly lead to a breaking change.

https://github.com/iqlusioninc/crates/blob/51d0b1e52ec726758d1b1baf8d81a924fc0c4a24/zeroize/src/lib.rs#L316

Edit:

The following types allow greater than `isize::MAX` allocations according to the API. They can have any length but 0, with `from_slice()` and `0..u32::MAX` in length for `generate()`.

- orion::aead::SecretKey
- orion::kdf::Password
- orion::kdf::Salt

The `zeroize` documentation also states that `Vec<>` never actually will allocate more than `isize::MAX`. Based on this, using `generate()` with `length > i32` would fail the length check in `zeroize` (and allocation beforehand according to `zeroize` docs) on a 32-bit target since there `isize::MAX == i32::MAX`. 

Using `from_slice` requires there to already be an allocation, so that should not be possible to trigger because the above newtypes use `Vec<u8>` and the allocation would have failed before interacting with Orion. Nonetheless, for consistency, I think both `from_slice` and `generate()` should limit lengths to a maximum of `isize::MAX`. This will be a breaking change.

**TL;DR**
The above newtypes must not accept input in `from_slice()` or lengths in `generate()` that are more than `isize::MAX`. Link to this issue in the code documentation for these two functions. This is a breaking change.
